biloba/config.py

```diff
@@ -99,6 +101,38 @@ def get_key(config, key, default=missing):
     return val
 
 
+def set_value(config, key, value):
+    """
+    Uses a dotted notation to set a value in a dict.
+
+    Example:
+        my_config = {
+            'http': {
+                'address': '127.0.0.1'
+            }
+        }
+
+        set_value(my_config, 'http.port', 5000)
+        set_value(my_config, 'log_level', 'debug')
+
+        my_config == {
+            'http': {
+                'address': '127.0.0.1',
+                'port': 5000
+            },
+            'log_level': 'debug'
+        }
+    """
+    try:
+        path, key = key.rsplit('.', 1)
+    except ValueError:
+        return config.__setitem__(key, value)
```

Comment:
Return value on a ``__setitem__`` doesn't make any sense - should this be on a separate line?

Also - ``config[key] = value`` is probably more pythonic?